modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearTxLocal.java

```diff
@@ -2455,6 +2455,11 @@ private boolean enlistWriteEntry(GridCacheContext cacheCtx,
                         }
 
                         if (readRepairStrategy != null) { // Checking and repairing each locked entry (if necessary).
+                            AffinityTopologyVersion topVer = topologyVersionSnapshot();
+
+                            if (topVer == null)
+                                topVer = entryTopVer;
```

Comment:
There is no docs for `entryTopVer` in the method signature. Why do you check it after `topologyVersionSnapshot()`? What is a full priority queue of topology versions?

Reply:
Rolled back this change since it cannot be finished/proved during this fix.